.gitignore

```diff
@@ -26,6 +26,7 @@ builder/rootfs/usr/bin
 cache/image/bin/
 client/dist/
 client/makeself/
+client-go/deis
```

Comment:
My original thought was to have it written in parallel like @Joshua-Anderson is doing here so that we could begin asking users to test it before we make it the official client, but @mboersma made the point that our CI runs will fully exercise the client and validate it pretty effectively. I'd like to see what he thinks about swapping them out in one fell swoop.